### ISSUE TITLE
new getting started experience for ev3dev-stretch

### ISSUE THREAD

**Issue by dlech** [ev3dev/ev3dev#952]
I've started a rough draft of what I think the new getting started page for ev3dev-stretch should look like based on feedback we've received over the years.

https://github.com/ev3dev/ev3dev/blob/ev3dev-stretch/docs/getting-started-ev3.rst

Comments, suggestions, pull requests, etc., please.

**Comment by WasabiFan** [comment]
Question: Do you intend to put this on ev3dev.org or RTD? I would vote for ev3dev.org because we have formatting control for download flow and visually splitting tracks, but I'm open to other options. My thinking is that technical docs should be on RTD and UX-focused ones should be on ev3dev.org.

**Comment by WasabiFan** [comment]
> TODO: make a list of supported Wi-Fi dongles somewhere

http://www.ev3dev.org/docs/networking/#with-a-wi-fi-dongl

Otherwise, looks good -- It'd be great if we could actually make set-up that simple :laughing: I'm interested to see what you can do there.

**Comment by jabrena** [comment]
Today, I will test the extension for Java:
https://marketplace.visualstudio.com/items?itemName=georgewfraser.vscode-javac

**Comment by jabrena** [comment]
Hi @dlech, @WasabiFan ,

When in the extesion:
https://github.com/ev3dev/vscode-hello-python#readme

appear this file:

``` 
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Download and Run",
            "type": "ev3devBrowser",
            "request": "launch",
            "program": "/home/robot/${workspaceRootFolderName}/hello.py"
        }
    ]
}
``` 

When the extesion use the attribute `"request": "launch",` is it using the new command-line tool: `brickrun`?
https://github.com/ev3dev/ev3dev/issues/790

Juan Antonio

**Comment by JorgePe** [comment]
That part of installing ev3dev with Etcher... is it because Etcher is available for all major OS's?
I just use gnome-disk-utility ("Disks" in Ubuntu).

**Comment by dlech** [comment]
> Do you intend to put this on ev3dev.org or RTD?

I'm leaning towards having most docs on RTD, mainly because of the versioning capability. The download page will stay on ev3dev.org for sure though.

**Comment by dlech** [comment]
> When the extesion use the attribute "request": "launch", is it using the new command-line tool: brickrun?

Yes, it is.

**Comment by dlech** [comment]
> That part of installing ev3dev with Etcher... is it because Etcher is available for all major OS's?

Yes. It lets us have one set of instructions that works on all OS's. And also, it uses bmaps, which speeds up flashing significantly by skipping all of the parts of the disk image that are just 0s.

**Comment by dwalton76** [comment]
> MicroSDHC memory card (cards > 32GB will not work

What's the scoop on the 32GB limit? I have a 128GB card in my EVB, everything with the filesystem seems to be ok.

**Comment by dlech** [comment]
The page is for EV3 only. The EV3 hardware is only capable of [SDHC](https://en.wikipedia.org/wiki/Secure_Digital#SDHC), which means 32GB max. Based on feedback from ev3dev users, cards larger than that may seem to sort of work, but have issues with random data corruption.

I will add additional pages for Raspberry Pi and BeagleBone, which may have different requirements.

**Comment by dlech** [comment]
According to [this](https://groups.google.com/forum/#!topic/beagleboard/rK3y2ih1kig) BeagleBone only officially supports SDHC as well, but SDXC will work albeit much slower than it should. So, our official docs will still say 32GB max for BeagleBone as well.

**Comment by dwalton76** [comment]
Gotcha.  I have a 128G in an EV3 too but it is still running jessie.  I haven't noticed issues with either but I haven't done tons with the EVB lately.

**Comment by dlech** [comment]
Pretty version with additional platforms: http://docs.ev3dev.org/en/ev3dev-stretch/index.html

**Comment by JorgePe** [comment]
Nice style sheet.
I like the way you are structuring the information.
And the tab tip is a good idea, it gives a fresh look... I started using tabs 2 years ago and now even write on it to know what is inside the SD card.
Perhaps people have more tips? Like using a wall charger and a LiPO battery (or fake batteries) as an UPS for long sessions.

**Comment by dwalton76** [comment]
Looks nice on my iPhone

**Comment by BertLindeman** [comment]
Looks good.

Is it helpful to "edit on github" for `http://docs.ev3dev.org/en/ev3dev-stretch/index.html`?
Some small typo's like "Rapsberry" and the lego site is not https so the link issues an error message.

**Comment by dlech** [comment]
Yes, that would be very helpful.

**Comment by BertLindeman** [comment]
Done ;-)

**Comment by dlech** [comment]
Stretch docs have been updated with more detailed platform-specific information: http://docs.ev3dev.org/en/ev3dev-stretch/platforms/index.html

**Comment by dlech** [comment]
Additional ideas: #734, #566, #492

**Comment by JorgePe** [comment]
I can't edit the Hardware Platforms so I add my suggestion here:
- Raspberry Pi Zero W also works with ev3dev and you can add it also to note 19 (built-in BT BLE) and 21, built-in Wi-Fi 

I didn't tried it yet with BrickPi3, just ev3dev by its own. Too bad it's so difficult to buy here in Portugal because I like it very much, can take it with me everywhere.
I see people using Arduino as a gateway to connect sensors to EV3 (I use the NodeMCU for the same reasons because it's small and has Wi-Fi) but I see the RpiZeroW as a much better "companion".

**Comment by dlech** [comment]
FYI: https://github.com/rtfd/readthedocs.org/issues/3203

If you delete "origin" from the link, you should be able to edit pages.

**Comment by dlech** [comment]
> Raspberry Pi Zero W also works with ev3dev and you can add it also to note 19 (built-in BT BLE) and 21, built-in Wi-Fi

I added Zero W to the table, but it looks like it was already mentioned in notes 19 and 21.

**Comment by JorgePe** [comment]
I tried to edit the table but the preview showed nothing so I canceled. Do you edit the table "by hand" or do you use some editor? The table in markup is quite a mess for someone without experience (i.e. me).

**Comment by dlech** [comment]
By hand. It helps to have a text editor with fixed width font, no wrapping and multiple cursors. I'm sure it is nearly impossible in the GitHub online editor.
